src/utils/bundleAssemblyHelpers.ts

```diff
@@ -43,31 +54,61 @@ export async function getResourcesReferenced(
   return DB.all('SELECT reference_id FROM "local_references"  WHERE origin_resource_id = ?', resourceId);
 }
 
+/**
+ * Uses SQL query to find all resource ids which correspond to fhir resources that
+ * reference the passed in object
+ * @param DB sqlite database object
+ * @param resourceId id of the resource we want references of
+ * @returns an array containing the resourceId, wrapped in an object,
+ * for each resource which references the resource with the passed in resourceId
+ */
+export async function getResourcesThatReference(
+  DB: sqlite.Database,
+  resourceId: string
+): Promise<{ origin_resource_id: string }[]> {
+  return DB.all('SELECT origin_resource_id FROM "local_references"  WHERE reference_id = ?', resourceId);
+}
+
 /**
  * Recursively searches through the graph of references to pull the ids of all resources
  * related through some chain of references to the resource with the passed in id
  * @param DB sqlite database object
  * @param resourceId the resource id to begin the recursion with
- * @param explored set of all previously processed resources (to avoid repeated effort)
+ * @param explored set of all previously processed resources within this recursive call(to avoid repeated effort)
```

Comment:
I called them entered resources since the property they're being added to on the bundle bundle is called entry, but if added is clearer to you then probably clearer to the rest of the team too. I agree that the variables are rough in these files

Reply:
okay that makes sense, I'm fine keeping them as enteredResources